src/Agent.Listener/CapabilitiesProvider.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener
+{
+    [ServiceLocator(Default = typeof(CapabilitiesProvider))]
+    public interface ICapabilitiesProvider : IAgentService
+    {
+        Task<Dictionary<string, string>> GetCapabilitiesAsync(string agentName, CancellationToken token);
+    }
+
+    public class Capability
+    {
+        public Capability(string name, string tool = null, string[] paths = null)
```

Comment:
i don't think `tool=null` is a good default value for constructor. if tool is null we will get empty as capability's value.